include/swift/AST/Decl.h

```diff
@@ -1687,6 +1687,10 @@ class ExtensionDecl final : public GenericContext, public Decl,
 
   MutableArrayRef<TypeLoc> Inherited;
 
+  /// This serves as a temporary place to check whether the extension was
+  /// parsed with its own unique gp list.
+  GenericParamList *ParsedGP;
```

Comment:
@CodaFi is there a better way to temporarily store this?

Reply:
Currently, we clone the generic parameters of the extended nominal into the field provided by `GenericContext`.  You can definitely look into just using that field.  If it's not set, `GenericParamListRequest` will run and just do the old thing.

Reply:
Please look at my latest commit and let me know what you think.